crypto/bn/asm/rsaz-2k-avxifma.pl

```diff
@@ -0,0 +1,856 @@
+# Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+# Copyright (c) 2024, Intel Corporation. All Rights Reserved.
+#
+# Licensed under the Apache License 2.0 (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+#
+# Written by Zhiguo Zhou <zhiguo.zhou@intel.com> and Wangyang Guo <wangyang.guo@intel.com>.
+# Special thanks to Tomasz Kantecki <tomasz.kantecki@intel.com> for his valuable suggestions.
+#
+# October 2024
+#
+# Initial release.
+#
+
+# $output is the last argument if it looks like a file (it has an extension)
+# $flavour is the first argument if it doesn't look like a file
+$output = $#ARGV >= 0 && $ARGV[$#ARGV] =~ m|\.\w+$| ? pop : undef;
+$flavour = $#ARGV >= 0 && $ARGV[0] !~ m|\.| ? shift : undef;
+
+$win64=0; $win64=1 if ($flavour =~ /[nm]asm|mingw64/ || $output =~ /\.asm$/);
+$avxifma=0;
+
+$0 =~ m/(.*[\/\\])[^\/\\]+$/; $dir=$1;
+( $xlate="${dir}x86_64-xlate.pl" and -f $xlate ) or
+( $xlate="${dir}../../perlasm/x86_64-xlate.pl" and -f $xlate) or
+die "can't locate x86_64-xlate.pl";
+
+# TODO: Find out the versions of NASM and Clang that supports VEX-encoded AVX-IFMA instructions
```

Comment:
In order to asses the support of `avxifma` instructions in various tool chains I did run builds with your changes where `$avxifma` is always set to 1. The result can be seen in this draft PR here:
 https://github.com/openssl/openssl/pull/25751

I'm sure your code gets compiled by clang 16. I have not looked at what clang version avxifma mnemonics appeared for the first time.

I'm looking at nasm support now.

The current check which enables the avxifma code is too strict IMO.